src/transformStateWithClones.js

```diff
@@ -7,7 +7,24 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const arrayStates = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(newState, action.extraData)
+    } else if (action.type === 'keysToRemove') {
+      for (const key of action.keysToRemove) {
+        delete newState[key];
+      }
+    } else {
+      newState = {};
```

Comment:
The 'clear' action type is missing. You need to handle the case where 'action.type === 'clear'' to set the state to an empty object.